lib/main.dart

```diff
@@ -98,3 +102,30 @@ Future<void> main() async {
     });
   }
 }
+
+Future<admin.App> _initializeAdmin() async {
+  final adminObj = admin.FirebaseAdmin.instance;
+  final tempDir = await getTemporaryDirectory();
+  final file = File('${tempDir.path}/service-account.json');
+  final data = jsonEncode({
+    "type": "service_account",
+    "project_id": "app-2i2i",
+    "private_key_id": "374c16c7dd0cd1df705fb4ef65bbbecfe8965ab4",
+    "private_key":
+        "-----BEGIN PRIVATE KEY-----\nMIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDSVdQwYlgp5yth\noLSTUCitVqQWnXN4/9hv/WLOIJ7B4FNkCTIqJyAM5bSjxzax0LnX+1rONyqGcc8c\nxR3gX3ANtLA63j1JmdhuaIccogXKzF24EMfIK6jCIBmFvNAcusCdDvUFCLb0nEDi\nhAie1jL00UFDFHFiKF/p/Q2zxMbLuNYguv6sVYZI9KFrJjd8QE9GPZ4Z4fY4rsMo\n8uGR2L/wjA/9WriSdAGsfMyzDe1fULhVqTP0K9LAmNw+ilbdEAOg6RkpqIzZXFbF\nLE566E/UrBtP5QNTogsTzUm+C/Nrh4Yz+B1WdN/OX7FLfIu2y9XxHj9Id1LaS94B\nO16aquIjAgMBAAECggEAOgmTzKv+PsW6OyyQVwupxbaf/VuBxtP1wWUN8+mdVMrU\nx8gfzvDPz688amqoRWVeOmHxevBO7B0QNJSRe11qOmXusLes+peYWFLAYV61sXE1\ntPnwXmPpo3MfB7+DP0Iqrsu3QSXX8eQBpV8gT8+z7MimtN85sAeK+7InK0DzR7ff\nRm2aEZQ/xNXPG4wLE6+IEd1OGpFXp0sMbRMcxuEId9nUKnU+uRWkT1cVAWNrJKO5\nqfdzRruzHmjp2CyqZwhMyZjprWzGfo07Gokf+ugU1+8loHhq9DE+cJsaVpjcG9Yb\nLere8jQ3EQJZVkW1gsyeda3wzY8+rF8LI7stplbraQKBgQDuhbbmQu2Wxv9P7ibE\n0X5fiJEX1Wt+3faFkF9tHmnZuYeo/3LuyRBGvs31+Wk6cK9nNmOE5AA5LJkS+r85\nLM8zvkx88jVcL+uprm0x4OqZEh5QR+phnldDSdr6jBCVLanhQD3rInx5laF5APCu\n3iQel5PEC6ozRhTyXWZTLamDOQKBgQDhv18r7IpsT4Ry7Zb/xrq4WQlpw9gboAUt\nqTsK0Jmj25NpW+Yr/Gs00t1qf64wwwei2i/F7GPU8vtqUMn7e8PP76sYdsBsG5Er\nmFi1yDRKe74xh69FNSwqPiLEnSpZztGanhC5DqiPQiFKJBF0PIGshApSh8RAQCn9\nAZRYAzLEOwKBgAyqXQPtGeCfwH7mDnj1Btjbz5iMZKDf+G8vM2H/8270QxfeOKQl\nWQ/oodcl30iTIz0/zhkIYlqm25n+ZpkKoBYHTNh1pA+5G7Ju6K11W/+0zpdEulVk\nqw2PpmkXdun0+shTOPZ7ZlCueVyLiDxA33ogYmBOnkMKvaIBA6X3DsFBAoGBAIFu\nHpvpb+fvo2ndbPDTOSUO74WzYslb8vweBhsKRLyc3STQKuTXQYQ0zfwMnouYll05\nqaBaC1cBxUJMdbH5YOhsanWJSIPzPkV3gI0g71IxJnfxan1MojjKDVcbITgCi0yS\nCdaAYAZbDQkcKuGw+0w1HFH7Q3rvDuPfrvTTSJ0XAoGADuQMeunmgG2C/FImLDsQ\nin36z3AJt7KgdWPcqQtunGdc1flWcd1Ryr5NcRhdkpcXyoCTXRiqxCoT/2kx0nGF\nYwfJazxpz2EdywQaKKPz9LfILx9k+qdYEEJN2INhjDXpqSomwzUJows8R3P1PtkA\nq9sJZyzHiJB65tu2V9kWUUA=\n-----END PRIVATE KEY-----\n",
```

Comment:
private key in code? that is never good

Reply:
We need function for it. We need a firebase function for generate custom token for firebase auth.